### ISSUE TITLE
`?url` CSS imports used in `links` are not bundled correctly in production

### ISSUE THREAD

**Issue by pcattori** [remix-run/remix#7786]
### What version of Remix are you using?

v0.0.0-nightly-0c99349-20231027

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

```tsx
// app/root.tsx
import tailwindHref from "~/tailwind.css?url";

export const links = () => [
  { rel: "stylesheet", href: tailwindHref },
];
```

### Expected Behavior

Processed styles show up in production

### Actual Behavior

Processed styles only show up in development

### Related Vite issues

- https://github.com/vitejs/vite/issues/2522
- https://github.com/vitejs/vite/issues/7842
- https://github.com/vitejs/vite/issues/13416

**Comment by pcattori** [comment]
Workaround: use side-effect `import`s instead of deprecated `?url` imports:

```diff
// app/root.tsx
- import { cssBundleHref } from "@remix-run/css-bundle";

- import tailwindHref from "~/tailwind.css?url";
+ import "~/tailwind.css";

- export const links = () => [
-   ...(cssBundleHref ? [{ rel: "stylesheet", href: cssBundleHref }] : []),
-   { rel: "stylesheet", href: tailwindHref },
- ];
```

**Comment by markdalgleish** [comment]
I think this is the same issue, logged against Vite: https://github.com/vitejs/vite/issues/13416

**Comment by 684efs3** [comment]
If I use the side-effect method (+ import "~/tailwind.css";), when I save in my editor, VS Code, all of the CSS will be removed from the localhost site. 

![Screenshot 2023-10-30 at 12 25 44 PM](https://github.com/remix-run/remix/assets/48864759/b75b28a1-645d-4526-9e1e-24238e7577ef)

If I hit [Command] [R] to refresh the browser, then the CSS styles will be applied. 

![Screenshot 2023-10-30 at 12 26 12 PM](https://github.com/remix-run/remix/assets/48864759/fae1b820-bece-4bdc-bccd-f1133475643d)

**Comment by markdalgleish** [comment]
@684efs3 Could you please share a repro?

**Comment by c43721** [comment]
I'm experiencing the same issue as @684efs3 but I don't have a reproduction. I'll edit this if I can get it repro'd on a new vite project with tailwind.

**Comment by pcfreak30** [comment]
> Workaround: use side-effect `import`s instead of `?url` imports:
> 
> ```diff
> // app/root.tsx
> - import { cssBundleHref } from "@remix-run/css-bundle";
> 
> - import tailwindHref from "~/tailwind.css?url";
> + import "~/tailwind.css";
> 
> - export const links = () => [
> -   ...(cssBundleHref ? [{ rel: "stylesheet", href: cssBundleHref }] : []),
> -   { rel: "stylesheet", href: tailwindHref },
> - ];
> ```
> 
> The difference in how `?url` is treated in dev vs. prod seems to be a Vite-specific issue and not Remix related.

This works for me but breaks the Links component/makes it impossible to use it for imported CSS. A PR is needed to integrate them properly.

**Comment by markdalgleish** [comment]
@pcfreak30

> This is still broken in 2.4.1

Sorry, what's still broken exactly? 

> This works for me but breaks the Links component/makes it impossible to use it for imported CSS. A PR is needed to properly integrate them (link component and CSS side effects?).

Sorry, I'm not quite following. Do you have an example of what's breaking?

> I also cannot confirm if `@remix-run/css-bundle` does anything.

The `@remix-run/css-bundle` package is a no-op in Vite (the `cssBundleHref` export is always `undefined`) and can be safely removed.

**Comment by pcfreak30** [comment]
> @pcfreak30
> 
> > This is still broken in 2.4.1
> 
> Sorry, what's still broken exactly? If you're referring to the original issue, it's an upstream issue in Vite that won't be fixed by a Remix release. This issue is only here so that we have a single place to link to and track all related Vite issues.
> 
> > This works for me but breaks the Links component/makes it impossible to use it for imported CSS. A PR is needed to properly integrate them (link component and CSS side effects?).
> 
> Sorry, I'm not quite following. Do you have an example of what's breaking?
> 
> > I also cannot confirm if `@remix-run/css-bundle` does anything.
> 
> The `@remix-run/css-bundle` package is a no-op in Vite (the `cssBundleHref` export is always `undefined`) and can be safely removed.

IIRC, the `?url`. It doesn't import it properly.

In dev `?url` works, in prod it encodes as `data:application/octet-stream;base64,`

I can only use side-effects and have vite bundle for me.

**Comment by pcfreak30** [comment]
It looks like the PR https://github.com/vitejs/vite/pull/15259 solves it, but it hasn't been merged yet.

**Comment by Danones** [comment]
> If I use the side-effect method (+ import "~/tailwind.css";), when I save in my editor, VS Code, all of the CSS will be removed from the localhost site.
> 
> ![Screenshot 2023-10-30 at 12 25 44 PM](https://private-user-images.githubusercontent.com/48864759/279127972-b75b28a1-645d-4526-9e1e-24238e7577ef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNzMzMjgsIm5iZiI6MTcwNDI3MzAyOCwicGF0aCI6Ii80ODg2NDc1OS8yNzkxMjc5NzItYjc1YjI4YTEtNjQ1ZC00NTI2LTllMWUtMjQyMzhlNzU3N2VmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDA5MTAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1MTY4ZDgxM2U5ZDNhOGZkMGE5NDVhMmFjNDlmMGNhNjE5ZTU2MDI3NWJiYmQ1OTlkYjk5OWEzMmVkMGM3MTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.k01p-NjkdrpdPwU_19_Zv1O62iNr_3yHauccePJvakc)
> 
> If I hit [Command] [R] to refresh the browser, then the CSS styles will be applied.
> 
> ![Screenshot 2023-10-30 at 12 26 12 PM](https://private-user-images.githubusercontent.com/48864759/279128106-fae1b820-bece-4bdc-bccd-f1133475643d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNzMzMjgsIm5iZiI6MTcwNDI3MzAyOCwicGF0aCI6Ii80ODg2NDc1OS8yNzkxMjgxMDYtZmFlMWI4MjAtYmVjZS00YmRjLWJjY2QtZjExMzM0NzU2NDNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDA5MTAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkN2I1N2NkMTlmNmQyOTkxYjYwODUwYjI4MmViMjcwMDgyZGJmOTQ4OTdkNWZjZTY5Zjg3OTQ3ZTJiYmI4MTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QcV-1Z-tgM2TYetEv9B3gweZsiL19TqcC6m-ScSbdDY)
> 
> This is a little annoying for developing.

After a couple of hours debugging, I reached the same issue with css.
This is preventing me from using Vite on my company app. I have decided to put the integration with vite on hold until this is fixed by Vite.

**Comment by pcfreak30** [comment]
> > If I use the side-effect method (+ import "~/tailwind.css";), when I save in my editor, VS Code, all of the CSS will be removed from the localhost site.
> > ![Screenshot 2023-10-30 at 12 25 44 PM](https://private-user-images.githubusercontent.com/48864759/279127972-b75b28a1-645d-4526-9e1e-24238e7577ef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNzMzMjgsIm5iZiI6MTcwNDI3MzAyOCwicGF0aCI6Ii80ODg2NDc1OS8yNzkxMjc5NzItYjc1YjI4YTEtNjQ1ZC00NTI2LTllMWUtMjQyMzhlNzU3N2VmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDA5MTAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1MTY4ZDgxM2U5ZDNhOGZkMGE5NDVhMmFjNDlmMGNhNjE5ZTU2MDI3NWJiYmQ1OTlkYjk5OWEzMmVkMGM3MTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.k01p-NjkdrpdPwU_19_Zv1O62iNr_3yHauccePJvakc)
> > If I hit [Command] [R] to refresh the browser, then the CSS styles will be applied.
> > ![Screenshot 2023-10-30 at 12 26 12 PM](https://private-user-images.githubusercontent.com/48864759/279128106-fae1b820-bece-4bdc-bccd-f1133475643d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNzMzMjgsIm5iZiI6MTcwNDI3MzAyOCwicGF0aCI6Ii80ODg2NDc1OS8yNzkxMjgxMDYtZmFlMWI4MjAtYmVjZS00YmRjLWJjY2QtZjExMzM0NzU2NDNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDA5MTAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkN2I1N2NkMTlmNmQyOTkxYjYwODUwYjI4MmViMjcwMDgyZGJmOTQ4OTdkNWZjZTY5Zjg3OTQ3ZTJiYmI4MTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QcV-1Z-tgM2TYetEv9B3gweZsiL19TqcC6m-ScSbdDY)
> > This is a little annoying for developing.
> 
> After a couple of hours debugging, I reached the same issue with css. This is preventing me from using Vite on my company app. I have decided to put the integration with vite on hold until this is fixed by Vite.

FYI you can make it work, it's just hacky.

**Comment by Danones** [comment]
> > > If I use the side-effect method (+ import "~/tailwind.css";), when I save in my editor, VS Code, all of the CSS will be removed from the localhost site.
> > > ![Screenshot 2023-10-30 at 12 25 44 PM](https://private-user-images.githubusercontent.com/48864759/279127972-b75b28a1-645d-4526-9e1e-24238e7577ef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNzMzMjgsIm5iZiI6MTcwNDI3MzAyOCwicGF0aCI6Ii80ODg2NDc1OS8yNzkxMjc5NzItYjc1YjI4YTEtNjQ1ZC00NTI2LTllMWUtMjQyMzhlNzU3N2VmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDA5MTAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1MTY4ZDgxM2U5ZDNhOGZkMGE5NDVhMmFjNDlmMGNhNjE5ZTU2MDI3NWJiYmQ1OTlkYjk5OWEzMmVkMGM3MTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.k01p-NjkdrpdPwU_19_Zv1O62iNr_3yHauccePJvakc)
> > > If I hit [Command] [R] to refresh the browser, then the CSS styles will be applied.
> > > ![Screenshot 2023-10-30 at 12 26 12 PM](https://private-user-images.githubusercontent.com/48864759/279128106-fae1b820-bece-4bdc-bccd-f1133475643d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNzMzMjgsIm5iZiI6MTcwNDI3MzAyOCwicGF0aCI6Ii80ODg2NDc1OS8yNzkxMjgxMDYtZmFlMWI4MjAtYmVjZS00YmRjLWJjY2QtZjExMzM0NzU2NDNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDA5MTAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkN2I1N2NkMTlmNmQyOTkxYjYwODUwYjI4MmViMjcwMDgyZGJmOTQ4OTdkNWZjZTY5Zjg3OTQ3ZTJiYmI4MTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QcV-1Z-tgM2TYetEv9B3gweZsiL19TqcC6m-ScSbdDY)
> > > This is a little annoying for developing.
> > 
> > 
> > After a couple of hours debugging, I reached the same issue with css. This is preventing me from using Vite on my company app. I have decided to put the integration with vite on hold until this is fixed by Vite.
> 
> FYI you can make it work, it's just hacky.

How? Curious to see how hacky it is

**Comment by pcfreak30** [comment]
> > > > If I use the side-effect method (+ import "~/tailwind.css";), when I save in my editor, VS Code, all of the CSS will be removed from the localhost site.
> > > > ![Screenshot 2023-10-30 at 12 25 44 PM](https://private-user-images.githubusercontent.com/48864759/279127972-b75b28a1-645d-4526-9e1e-24238e7577ef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNzMzMjgsIm5iZiI6MTcwNDI3MzAyOCwicGF0aCI6Ii80ODg2NDc1OS8yNzkxMjc5NzItYjc1YjI4YTEtNjQ1ZC00NTI2LTllMWUtMjQyMzhlNzU3N2VmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDA5MTAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1MTY4ZDgxM2U5ZDNhOGZkMGE5NDVhMmFjNDlmMGNhNjE5ZTU2MDI3NWJiYmQ1OTlkYjk5OWEzMmVkMGM3MTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.k01p-NjkdrpdPwU_19_Zv1O62iNr_3yHauccePJvakc)
> > > > If I hit [Command] [R] to refresh the browser, then the CSS styles will be applied.
> > > > ![Screenshot 2023-10-30 at 12 26 12 PM](https://private-user-images.githubusercontent.com/48864759/279128106-fae1b820-bece-4bdc-bccd-f1133475643d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNzMzMjgsIm5iZiI6MTcwNDI3MzAyOCwicGF0aCI6Ii80ODg2NDc1OS8yNzkxMjgxMDYtZmFlMWI4MjAtYmVjZS00YmRjLWJjY2QtZjExMzM0NzU2NDNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDA5MTAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkN2I1N2NkMTlmNmQyOTkxYjYwODUwYjI4MmViMjcwMDgyZGJmOTQ4OTdkNWZjZTY5Zjg3OTQ3ZTJiYmI4MTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QcV-1Z-tgM2TYetEv9B3gweZsiL19TqcC6m-ScSbdDY)
> > > > This is a little annoying for developing.
> > > 
> > > 
> > > After a couple of hours debugging, I reached the same issue with css. This is preventing me from using Vite on my company app. I have decided to put the integration with vite on hold until this is fixed by Vite.
> > 
> > 
> > FYI you can make it work, it's just hacky.
> 
> How? Curious to see how hacky it is

https://git.lumeweb.com/LumeWeb/web3.news/src/commit/c1897e4d7c225a5511cadaf2711fbe9ef353d861/app/root.tsx#L12

**Comment by Danones** [comment]
> > > > > If I use the side-effect method (+ import "~/tailwind.css";), when I save in my editor, VS Code, all of the CSS will be removed from the localhost site.
> > > > > ![Screenshot 2023-10-30 at 12 25 44 PM](https://private-user-images.githubusercontent.com/48864759/279127972-b75b28a1-645d-4526-9e1e-24238e7577ef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNzMzMjgsIm5iZiI6MTcwNDI3MzAyOCwicGF0aCI6Ii80ODg2NDc1OS8yNzkxMjc5NzItYjc1YjI4YTEtNjQ1ZC00NTI2LTllMWUtMjQyMzhlNzU3N2VmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDA5MTAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1MTY4ZDgxM2U5ZDNhOGZkMGE5NDVhMmFjNDlmMGNhNjE5ZTU2MDI3NWJiYmQ1OTlkYjk5OWEzMmVkMGM3MTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.k01p-NjkdrpdPwU_19_Zv1O62iNr_3yHauccePJvakc)
> > > > > If I hit [Command] [R] to refresh the browser, then the CSS styles will be applied.
> > > > > ![Screenshot 2023-10-30 at 12 26 12 PM](https://private-user-images.githubusercontent.com/48864759/279128106-fae1b820-bece-4bdc-bccd-f1133475643d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyNzMzMjgsIm5iZiI6MTcwNDI3MzAyOCwicGF0aCI6Ii80ODg2NDc1OS8yNzkxMjgxMDYtZmFlMWI4MjAtYmVjZS00YmRjLWJjY2QtZjExMzM0NzU2NDNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDA5MTAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkN2I1N2NkMTlmNmQyOTkxYjYwODUwYjI4MmViMjcwMDgyZGJmOTQ4OTdkNWZjZTY5Zjg3OTQ3ZTJiYmI4MTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QcV-1Z-tgM2TYetEv9B3gweZsiL19TqcC6m-ScSbdDY)
> > > > > This is a little annoying for developing.
> > > > 
> > > > 
> > > > After a couple of hours debugging, I reached the same issue with css. This is preventing me from using Vite on my company app. I have decided to put the integration with vite on hold until this is fixed by Vite.
> > > 
> > > 
> > > FYI you can make it work, it's just hacky.
> > 
> > 
> > How? Curious to see how hacky it is
> 
> https://git.lumeweb.com/LumeWeb/web3.news/src/commit/c1897e4d7c225a5511cadaf2711fbe9ef353d861/app/root.tsx#L12

For me it does not work. Importing the styles as a side effect is something that I am already doing as part of the latest remix release and vite support migration guide. 

I noticed that on your code, you are importing the `global.css`  directly in the root, and I have tried doing the same but the issue remains. 

Maybe I misunderstood your point, but this does not seem to work

**Comment by pcattori** [comment]
https://github.com/vitejs/vite/pull/15259 was merged earlier today

**Comment by markdalgleish** [comment]
Vite's fix for this is currently available in v5.1.0-beta.0: https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md#510-beta0-2024-01-15

**Comment by pcattori** [comment]
Closing as Vite v5.1 has shipped.
